source/vulkan/ht_vkswapchain.cpp

```diff
@@ -0,0 +1,271 @@
+
+                for ( std::vector<VkSurfaceFormatKHR>::iterator sfor = surfaceFormats.begin(); sfor != surfaceFormats.end(); ++sfor)
+                {
+                    if (( sfor->format == VK_FORMAT_UNDEFINED ) || ( sfor->format == choosenSurfaceFormat.format && sfor->colorSpace == choosenSurfaceFormat.colorSpace ))
+                    {
+                        choosenFormatFound = true;
+                        break;
+                    }
+                }
+
+                choosenSurfaceFormat = surfaceFormats[0];
+
+                /*
+                * We need to discuss the policy to choose the present modes yet
+                */
+                VkPresentModeKHR choosenPresentMode = VK_PRESENT_MODE_FIFO_KHR;
+                bool presentModeFound = false;
+
+                for ( std::vector<VkPresentModeKHR>::iterator prsnt = presentModes.begin(); prsnt != presentModes.end(); ++prsnt)
+                {
+                    if ( *prsnt == VK_PRESENT_MODE_MAILBOX_KHR )
+                    {
+                        choosenPresentMode = VK_PRESENT_MODE_MAILBOX_KHR;
+                        presentModeFound = true;
+                        break;
+                    }
+
+                    if ( *prsnt == VK_PRESENT_MODE_FIFO_KHR )
+                        presentModeFound = true;
+                }
+
+                if ( !presentModeFound )
+                {
+                    HT_ERROR_PRINTF("VKSwapChain::Initialize(): Default present mode not found.\n");
+                    return false;
+                }
+
+                VkSurfaceCapabilitiesKHR surfaceCapabilities;
+                vkGetPhysicalDeviceSurfaceCapabilitiesKHR(device, m_surface, &surfaceCapabilities); 
+
+                uint32_t imageCount = surfaceCapabilities.minImageCount + 1 <= surfaceCapabilities.maxImageCount ?
+                    surfaceCapabilities.minImageCount + 1 : surfaceCapabilities.maxImageCount;
+
+                VkExtent2D choosenExtent;
+
+                if ( surfaceCapabilities.currentExtent.width != UINT32_MAX )
+                    choosenExtent = surfaceCapabilities.currentExtent;
+                else
+                {
+                    choosenExtent.width = std::max(surfaceCapabilities.minImageExtent.width, std::min(surfaceCapabilities.maxImageExtent.width, m_width));
+                    choosenExtent.height = std::max(surfaceCapabilities.minImageExtent.height, std::min(surfaceCapabilities.maxImageExtent.height, m_height));
+                }
+
+                VkSwapchainCreateInfoKHR createSwapChainInfo;
+                createSwapChainInfo.sType = VK_STRUCTURE_TYPE_SWAPCHAIN_CREATE_INFO_KHR;
+                createSwapChainInfo.surface = m_surface;
+                createSwapChainInfo.minImageCount = imageCount;
+                createSwapChainInfo.imageColorSpace = choosenSurfaceFormat.colorSpace;
+                /* I do believe in the first version we are not goig to enable occulus, vive and others... */
```

Comment:
You are correct. At this moment, immediate support VR technology is not planned, but it will be supported eventually.